cmds/cbmem/main.go

```diff
@@ -0,0 +1,350 @@
+		TS_RO_PARAMS_INIT:               "RO parameter init",
+		TS_RO_VB_INIT:                   "RO vboot init",
+		TS_RO_VB_SELECT_FIRMWARE:        "RO vboot select firmware",
+		TS_RO_VB_SELECT_AND_LOAD_KERNEL: "RO vboot select&load kernel",
+		TS_RW_VB_SELECT_AND_LOAD_KERNEL: "RW vboot select&load kernel",
+		TS_VB_SELECT_AND_LOAD_KERNEL:    "vboot select&load kernel",
+		TS_VB_EC_VBOOT_DONE:             "finished EC verification",
+		TS_CROSSYSTEM_DATA:              "crossystem data",
+		TS_START_KERNEL:                 "start kernel",
+
+		// FSP related timestamps
+		TS_FSP_MEMORY_INIT_START:   "calling FspMemoryInit",
+		TS_FSP_MEMORY_INIT_END:     "returning from FspMemoryInit",
+		TS_FSP_TEMP_RAM_EXIT_START: "calling FspTempRamExit",
+		TS_FSP_TEMP_RAM_EXIT_END:   "returning from FspTempRamExit",
+		TS_FSP_SILICON_INIT_START:  "calling FspSiliconInit",
+		TS_FSP_SILICON_INIT_END:    "returning from FspSiliconInit",
+		TS_FSP_BEFORE_ENUMERATE:    "calling FspNotify(AfterPciEnumeration)",
+		TS_FSP_AFTER_ENUMERATE:     "returning from FspNotify(AfterPciEnumeration)",
+		TS_FSP_BEFORE_FINALIZE:     "calling FspNotify(ReadyToBoot)",
+		TS_FSP_AFTER_FINALIZE:      "returning from FspNotify(ReadyToBoot)",
+	}
+
+	mem                 = flag.String("mem", "/dev/mem", "file for coreboot image")
+	debug               = func(string, ...interface{}) {}
+	addr                int64
+	size                int
+	mf                  *os.File
+	console             bool
+	coverage            bool
+	list                bool
+	hexdump             bool
+	timestamps          bool
+	parseabletimestamps bool
+	verbose             bool
+	version             bool
+	dumpJSON            = flag.Bool("json", true, "Output tables in JSON format")
+)
+
+func init() {
+	flag.BoolVar(&console, "console", false, "print cbmem console")
+	flag.BoolVar(&coverage, "coverage", false, "dump coverage information")
+	flag.BoolVar(&list, "list", false, "print cbmem table of contents")
+	flag.BoolVar(&hexdump, "hexdump", false, "print hexdump of cbmem area")
+	flag.BoolVar(&timestamps, "timestamps", false, "print timestamp information")
+	flag.BoolVar(&parseabletimestamps, "parseable-timestamps", false, "print parseable timestamps")
+	flag.BoolVar(&verbose, "verbose", false, "verbose (debugging) output")
+	flag.BoolVar(&version, "version", false, "print the version")
+	flag.BoolVar(&console, "c", false, "print cbmem console")
+	flag.BoolVar(&coverage, "C", false, "dump coverage information")
+	flag.BoolVar(&list, "l", false, "print cbmem table of contents")
+	flag.BoolVar(&hexdump, "x", false, "print hexdump of cbmem area")
+	flag.BoolVar(&timestamps, "t", false, "print timestamp information")
+	flag.BoolVar(&parseabletimestamps, "T", false, "print parseable timestamps")
+	flag.BoolVar(&verbose, "v", false, "verbose (debugging) output")
+	flag.BoolVar(&version, "V", false, "print the version")
+}
+
+func parse_cbtable(mf io.ReaderAt, address int64, sz int) error {
```

Comment:
It would be nice if this function signature was more like 

`func parseCBTable(r io.ReaderAt, address int64, size int) (CBTable, error)`

instead of filling in a global cbtable

Reply:
done